src/core/ddsi/src/ddsi_cdrstream.c

```diff
@@ -3181,6 +3258,17 @@ static bool dds_stream_print_sample1 (char * __restrict *buf, size_t * __restric
       case DDS_OP_ADR: {
         if (!delimited_sz || (is->m_index - delimited_offs < delimited_sz))
         {
+          if (op_type_optional (insn))
+          {
+            bool present = is_mutable_member ? true : dds_is_get1 (is);
```

Comment:
I'd play around with creating the equivalent of `is_member_present` for the CDR stream — i.e., `!op_type_optional(insn) || is_mutable_member || dds_is_get1 (is)` — so that the logic here simplifies to `if (!is_member_present_cdr (is, insn)) { skip }`

Not sure if it really is beneficial, hence the "play around with" part.